sync/event_listener.go

```diff
@@ -0,0 +1,25 @@
+package sync
+
+import "time"
+
+type EventListener interface {
+	OnSyncStepDone(op string, blockNum uint64, took time.Duration)
+	OnReorg(blockNum uint64)
+}
+
+type SelectiveListener struct {
+	OnSyncStepDoneCb func(op string, blockNum uint64, took time.Duration)
```

Comment:
Can we have separate callbacks instead of having the `op` enum?

Reply:
We can, but we currently dont really have a use for it. Should be an easy refactor when we do.